miden-lib/asm/miden/kernels/tx/account.masm

```diff
@@ -147,12 +147,18 @@ end
 #! - value is the value to increment the nonce by. value can be at most 2^32 - 1 otherwise this
 #!   procedure panics.
 export.incr_nonce
-    # emit event to signal that account nonce is being incremented
-    emit.ACCOUNT_INCREMENT_NONCE_EVENT
-
     u32assert.err=ERR_ACCOUNT_NONCE_INCR_MUST_BE_U32
+
+    # duplicate the value to be able to emit an event after the nonce is being incremented
+    dup
+
     exec.memory::get_acct_nonce add
     exec.memory::set_acct_nonce
+
+    # TODO: we execute `push.1 drop` before `emit` as decorators are not supported without other
+    # instructions - see: https://github.com/0xPolygonMiden/miden-vm/issues/1122
+    # emit event to signal that account nonce is being incremented
+    push.1 drop emit.ACCOUNT_INCREMENT_NONCE_EVENT drop
```

Comment:
Should we do this right after `u32assert` (i.e., instead of `dup`)? As far as I can tell, that's the only place where this can fail.

Reply:
Indeed, the rest of the function is infallible! I'll move this `emit` to the `dup`'s place.